core/schema/modulesource.go

```diff
@@ -27,10 +31,10 @@ type moduleSourceArgs struct {
 }
 
 func (s *moduleSchema) moduleSource(ctx context.Context, query *core.Query, args moduleSourceArgs) (*core.ModuleSource, error) {
-	parsed := parseRefString(args.RefString)
-	modPath, modVersion, hasVersion, isGitHub := parsed.modPath, parsed.modVersion, parsed.hasVersion, parsed.isGitHub
+	parsed, err := parseRefString(ctx, query.Buildkit, args.RefString)
```

Comment:
we're not checking of error after this? seems like we need to revisit the error handling of this `parseRefString` function as it seems like we might want to a specific error in some cases?

Reply:
Reverted this change, as said above 🙏